### PR TITLE
Reorganize how DefaultSettings are loaded

### PR DESCRIPTION
**Pull request by mwjames**
This PR is made in reference to: # [0]

This PR addresses or contains:

- Only `"SemanticMediaWiki.php"` is autoloaded in `composer.json`
- `DefaultSettings.php` returns an array and is assigned in `SemanticMediaWiki::load` to $GLOBALS and can as in case of [0] be modified to adhere custom specific processing 

This PR includes:
- [ ] Tests (unit/integration)
- [x] CI build passed

[0] https://github.com/Wikia/app/pull/11922

### CODE REVIEW

File: DefaultSettings.php [hunk 1]
```diff
@@ -16,1184 +16,1210 @@
...
-###
-# Settings for inline queries ({{#ask:...}}) and for semantic queries in
-# general. This can especially be used to prevent overly high server-load due
-# to complex queries. The following settings affect all queries, wherever they
-# occur.
-##
-$GLOBALS['smwgQEnabled'] = true;   // (De)activates all query related features and interfaces
-$GLOBALS['smwgQMaxLimit'] = 10000; // Max number of results *ever* retrieved, even when using special query pages.
-$GLOBALS['smwgIgnoreQueryErrors'] = true; // Should queries be executed even if some errors were detected?
+return array(
```
Comment (JeroenDeDauw) [inline DefaultSettings.php]: y u no `[]`

File: load.php [hunk 2]
```diff
@@ -8,26 +8,60 @@
...
+		include_once __DIR__ . '/src/Defines.php';
+		include_once __DIR__ . '/src/GlobalFunctions.php';
+
+		$settings = include __DIR__ . '/DefaultSettings.php';
+
+		foreach ( $settings as $key => $value ) {
+			$GLOBALS[$key] = $value;
+		}
+
+		unset( $settings );
```
Comment (JeroenDeDauw) [inline load.php]: huh... if you don't want it in scope, then simply don't put it in it in the first place. You can inline the include in the foreach predicate, or do that and factor the include into a getDefaultSettings function, though I think the later is overkill here

File: DefaultSettings.php (hunk 1 above)
Comment (mwjames) [inline DefaultSettings.php]: For now I rely on `array` to avoid another 2.4.2 back-port issue.

File: load.php (hunk 2 above)
Comment (mwjames) [inline load.php]: > the include into a getDefaultSettings function, though I think the later is overkill here

Wikia/app#11922 changed the settings file to get access to individual settings hence the anticipation of this by providing that in a form without someone having to edit DefaultSettings.

They have to make some changes eventually but I'd rather see them to do those changes in this file with less potential of conflict.

Comment (mwjames) [inline load.php]: >  You can inline the include in the foreach predicate,

I experimented with that but somehow the readability suffered.

Comment (JeroenDeDauw) [inline load.php]: Not sure you got what I suggested. Change here: https://github.com/SemanticMediaWiki/SemanticMediaWiki/pull/2035

**Pull request merged**